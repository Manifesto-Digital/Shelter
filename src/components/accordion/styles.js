import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
