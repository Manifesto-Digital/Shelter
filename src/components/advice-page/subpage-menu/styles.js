import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
