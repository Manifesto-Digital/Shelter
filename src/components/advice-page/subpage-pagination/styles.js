import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
