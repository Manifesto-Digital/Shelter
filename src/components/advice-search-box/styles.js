import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
