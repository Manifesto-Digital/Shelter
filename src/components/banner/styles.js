import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
