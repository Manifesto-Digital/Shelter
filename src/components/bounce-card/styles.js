import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
