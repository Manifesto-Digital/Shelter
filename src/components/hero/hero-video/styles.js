import styled from 'styled-components';
import ReactPlayer from 'react-player';
import { breakpoint } from '../../theme/breakpoint';

export const VideoHero = styled.section`
  overflow: hidden;
  padding-top: 60vh;
  position: relative;
`;

export const VideoWrapper = styled.div``;

export const ImageFallback = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background-image: url(${({ bgImage }) => bgImage});
  background-size: cover;
  background-position: center;
`;

export const Video = styled(ReactPlayer)`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
`;

export const TitleWrapper = styled.div`
  position: absolute;
  left: 50%;
  max-width: 520px;
  top: ${props => props.theme.spacing.xl};
  transform: translateX(-50%);
  width: 100%;
  padding: 0 ${props => props.theme.spacing.standard};
  text-align: center;
  color: ${props => props.theme.palette.white};
`;

export const Title = styled.h1`
  margin-bottom: 0;
  font-size: ${props => props.theme.headers.h1};

  ${breakpoint.tablet`
    font-size: ${props => props.theme.fontsize.hero};
  `}
`;

export const Subtitle = styled.p`
  text-align: center;
  font-size: ${props => props.theme.headers.h2};
`;

export const Red = styled.span`
  color: ${props => props.theme.palette.primary};
`;

export const HeroBanner = styled.div`
  background-color: ${props => props.theme.palette.overlay};
  bottom: 0;
  left: 0;
  padding: ${props => props.theme.spacing.standard};
  position: absolute;
  right: 0;
  width: 100%;
`;

export const ContentContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0 auto;
  width: 100%;
  max-width: 400px;
`;

export const BannerText = styled.p`
  color: ${props => props.theme.palette.white};
  margin-right: ${props => props.theme.spacing.standard};
  margin-bottom: 0;
  padding-left: ${props => props.theme.spacing.standard};
  border-left: 3px solid ${props => props.theme.palette.white};
  white-space: pre-wrap;

  ${breakpoint.tablet`
    margin-right: ${props => props.theme.spacing.large};
  `}
`;

export const StickyBanner = styled.div`
  background-color: ${props => props.theme.palette.primary};
`;
