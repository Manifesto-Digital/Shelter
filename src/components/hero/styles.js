import styled from 'styled-components'
import { breakpoint } from '../theme/breakpoint'
import { absoluteCenter } from '../styled/utils'

export const HeroNoCard = styled.div`
  width: 100vw;
  position: relative;
  overflow: hidden;
  padding-top: 350px;
  margin-bottom: ${props => props.theme.spacing.padding};
  ${breakpoint.tablet`
    padding-top: 60vh;
`};
`

export const MediaWrapper = styled.div`
  ${absoluteCenter};
  width: 100%;
  height: 100%;
  overflow: hidden;

  img {
    ${absoluteCenter};
    min-width: 100%;
    min-height: 100%;
    width: auto;
    height: auto;
    max-width: none;
    height: 350px;

    ${breakpoint.tablet`
        height: auto;
    `};
  }
`

export const Content = styled.div`
  ${absoluteCenter};
  text-align: center;
  width: 100%;
  padding: 0 ${props => props.theme.spacing.padding};
  color: ${props =>
    props.black
      ? `${props.theme.palette.black}`
      : `${props.theme.palette.white}`};
`

export const Title = styled.h1`
  margin-bottom: 0;
`
export const Subtitle = styled.p`
  font-size: ${props => props.theme.headers.h2};
  margin-bottom: 0;
`
