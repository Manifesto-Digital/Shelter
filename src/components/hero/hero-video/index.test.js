import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow } from 'enzyme';
import { mountWithTheme, snapshotComponent } from 'test-helpers';
import VideoHero from '.';
import {
  createFactory,
  createImage,
  createVideo,
  createExternalRef,
} from '../../../utils/test-factories';
import { Title, Video, ImageFallback } from './styles';

// Default props
export const createHeroVideo = createFactory({
  title: 'Urban Rush',
  subtitle: 'London',
  image: createImage(),
  video: createVideo(),
  buttonText: 'Button text',
  bannerText: '15 MILES\nLONDON\n06 OCT 2019',
  eventLink: createExternalRef(),
});

jest.mock('react-player', () => 'React-player');

it('renders correctly', () => {
  const mockData = createHeroVideo();
  snapshotComponent(<VideoHero {...mockData} />);
});

it('title is rendered', () => {
  const mockData = createHeroVideo({
    title: 'Urban Rush',
  });
  const wrapper = shallow(<VideoHero {...mockData} />);
  expect(wrapper.find(Title).text()).toEqual(mockData.title);
});

it('includes a link to the event', () => {
  const mockData = createHeroVideo({
    eventLink: createExternalRef({
      newTab: true,
      URL:
        'https://endurancecui.active.com/new/events/57227451/select-race?_p=9778819554116866&e4q=b4c582fe-43bc-41b0-8df3-221750f7842b&e4p=98898649-9a90-47cc-a977-0197a099719b&e4ts=1538665644&e4c=active&e4e=snawe00000000&e4rt=Safetynet&e4h=a74b16f1d10ebeed229cf4c8311a6f08&_ga=2.193888045.1618800628.1549375348-1259773413.1541001577',
    }),
  });

  const wrapper = mountWithTheme(<VideoHero {...mockData} />);

  expect(wrapper.find('a').prop('href')).toEqual(mockData.eventLink.URL);
});

it('renders an image fallback if there is an error loading the video', () => {
  const mockData = createHeroVideo({
    title: 'Urban Rush',
  });
  const wrapper = mountWithTheme(<VideoHero {...mockData} />);

  act(() => {
    wrapper.find(Video).prop('onReady')();
  });
  wrapper.update();

  expect(wrapper.find(ImageFallback)).toHaveLength(0);

  act(() => {
    wrapper
      .find(Video)
      .simulate('error', { target: { error: 'Error has occurred' } });
  });

  wrapper.update();

  expect(wrapper.find(ImageFallback)).toHaveLength(1);
});
