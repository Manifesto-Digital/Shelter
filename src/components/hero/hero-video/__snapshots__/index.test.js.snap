// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  overflow: hidden;
  padding-top: 60vh;
  position: relative;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background-image: url(//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg);
  background-size: cover;
  background-position: center;
}

.c1 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c3 {
  position: absolute;
  left: 50%;
  max-width: 520px;
  top: 60px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 100%;
  padding: 0 20px;
  text-align: center;
  color: #fff;
}

.c4 {
  margin-bottom: 0;
  font-size: 2.35em;
}

.c6 {
  text-align: center;
  font-size: 1.8em;
}

.c5 {
  color: #ea232d;
}

.c7 {
  background-color: rgba(43,46,56,0.7);
  bottom: 0;
  left: 0;
  padding: 20px;
  position: absolute;
  right: 0;
  width: 100%;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 auto;
  width: 100%;
  max-width: 400px;
}

.c9 {
  color: #fff;
  margin-right: 20px;
  margin-bottom: 0;
  padding-left: 20px;
  border-left: 3px solid #fff;
  white-space: pre-wrap;
}

.c10 {
  display: inline-block;
  text-align: center;
  padding: 10px 1em;
  font-weight: normal;
  border: none;
  border: 4px solid #fff;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.c10:hover {
  color: #212122;
  background-color: #fff;
}

.c11 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

@media (min-width:40.0625em) {
  .c4 {
    font-size: 4rem;
  }
}

@media (min-width:40.0625em) {
  .c9 {
    margin-right: 40px;
  }
}

<section
  className="c0"
>
  <React-player
    className="c1"
    config={
      Object {
        "file": Object {
          "attributes": Object {
            "poster": "data:image/gif,AAAA",
            "style": Object {
              "height": "auto",
              "width": "auto",
            },
          },
        },
      }
    }
    height={null}
    loop={true}
    muted={true}
    onError={[Function]}
    onReady={[Function]}
    playing={true}
    url="//videos.ctfassets.net/6sxvmndnpn0s/5hlfsk9dwVVNBlbpWHSLvK/9f1d647e6332be389465c9d014a57a26/shelter-urban-rush.mp4"
    width={null}
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  >
    <h1
      className="c4"
    >
      Urban
       
      <span
        className="c5"
      >
        Rush
      </span>
    </h1>
    <p
      className="c6"
    >
      London
    </p>
  </div>
  <div
    className="c7"
  >
    <div
      className="c8"
    >
      <p
        className="c9"
      >
        15 MILES
LONDON
06 OCT 2019
      </p>
      <a
        className="c10"
        href="https://endurancecui.active.com/new/events/57227451/select-race?_p=9778819554116866&e4q=b4c582fe-43bc-41b0-8df3-221750f7842b&e4p=98898649-9a90-47cc-a977-0197a099719b&e4ts=1538665644&e4c=active&e4e=snawe00000000&e4rt=Safetynet&e4h=a74b16f1d10ebeed229cf4c8311a6f08&_ga=2.193888045.1618800628.1549375348-1259773413.1541001577"
        rel="noopener"
        target="_self"
      >
        Button text
      </a>
    </div>
  </div>
  <div
    className="c11"
  />
</section>
`;
